Sources/Lifecycle/Lifecycle.swift

```diff
@@ -152,6 +147,15 @@ public struct ServiceLifecycle {
         self.underlying.wait()
     }
 
+    private func setupForStart() {
```

Comment:
we could consider an assertion here based on a static to make sure theses never called twice since that may lead to undefined behavior. wdyt?